application/src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/Transfer.java

```diff
@@ -31,16 +35,20 @@ public class Transfer {
 
   private final List<Edge> edges;
 
-  public Transfer(int toStop, List<Edge> edges) {
+  private final ImmutableSet<StreetMode> modes;
```

Comment:
I know there was a lot of back and forth with this. In the end @t2gran says he didn't want Guava collection, so lets use Collections.unmodifiableSet.

```suggestion
  private final Set<StreetMode> modes;
```

Reply:
The reason we can do it here is that it's not serialized.

Reply:
I changed it to this